packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -636,18 +680,25 @@ export async function continueStaticPrerender(
 type ContinueResumeOptions = {
   inlinedDataStream: ReadableStream<Uint8Array>
   getServerInsertedHTML: () => Promise<string>
+  getServerInsertedMetadata: () => Promise<string>
 }
 
 export async function continueDynamicHTMLResume(
   renderStream: ReadableStream<Uint8Array>,
-  { inlinedDataStream, getServerInsertedHTML }: ContinueResumeOptions
+  {
+    inlinedDataStream,
+    getServerInsertedHTML,
+    getServerInsertedMetadata,
+  }: ContinueResumeOptions
 ) {
   return (
     renderStream
       // Buffer everything to avoid flushing too frequently
       .pipeThrough(createBufferedTransformStream())
       // Insert generated tags to head
       .pipeThrough(createHeadInsertionTransformStream(getServerInsertedHTML))
+      // Insert generated metadata to body
+      .pipeThrough(createInsertedMetadataStream(getServerInsertedMetadata))
```

Comment:
If this prerender was empty then the initial chunks of the resume will contain the document. The way `createInsertedMetadataStream` is set up now this will lead to the metadata chunks being in front of the doctype I think